registry/storage/driver/filesystem/driver.go

```diff
@@ -282,10 +283,9 @@ func (d *driver) Delete(ctx context.Context, subPath string) error {
 	return err
 }
 
-// URLFor returns a URL which may be used to retrieve the content stored at the given path.
-// May return an UnsupportedMethodErr in certain StorageDriver implementations.
-func (d *driver) URLFor(ctx context.Context, path string, options map[string]interface{}) (string, error) {
-	return "", storagedriver.ErrUnsupportedMethod{}
+// RedirectURL returns a URL which may be used to retrieve the content stored at the given path.
+func (d *driver) RedirectURL(*http.Request, string) (string, error) {
+	return "", nil
 }
```

Comment:
Related to the above; could an alternative be to move `RedirectURL` to a separate interface, which would be an optional feature to be implemented by drivers.

Drivers not needing this feature could then omit implementing it.

Reply:
I had considered that, but [it's really awkward and verbose](https://blog.merovius.de/posts/2017-07-30-the-trouble-with-optional-interfaces/#they-remove-a-lot-of-the-power-of-interfaces) to make a wrapper struct type (read: storage middleware) which optionally implements an interface iff the inner type implements the same interface. Instead I tweaked the method's contract to make it less weird that it's mandatory.